backends/guest/include/common/util.h

```diff
@@ -52,6 +51,29 @@ struct signature_type_info {
 	size_t size;
 };
 
+/*
+ * Truncated form of a timestamp, fields are verbatim the same as in timestamp_t
+ */
+struct sv_trunc_timestamp {
+	leint16_t year;
+	uint8_t month;
+	uint8_t day;
+	uint8_t hour;
+	uint8_t minute;
+	uint8_t second;
+} __attribute__((packed));
+
```

Comment:
This is an incomplete definition of the timestamp struct.

Reply:
Please reuse the timestamp_t struct here.

Reply:
Already noted why `timestamp_t` will not work here.

BUT checkout the top of `external/libstb-secvar/src/pseries.c`:
```
struct var_timestamp
{
  leint16_t year;
  uint8_t month;
  uint8_t day;
  uint8_t hour;
  uint8_t minute;
  uint8_t second;
} SV_PACKED;

typedef struct var_timestamp var_timestamp_t;

struct signed_variable
{
  uint8_t version; /* must be 0 */
  var_timestamp_t time;
} SV_PACKED;
```

I think we should work on exporting this struct to a headerfile in `libstb-secvar`. That way we won't need to maintain our own implementation. I am not sure what the current status is of guest-sb support or if FW is already dedicated to a release of libstb-secvar so let me know if we should keep this commit for now and revert some of it later  (after these structs are available in a header file in libstb-sevar) or we should just hold off on applying this commit.

Reply:
I agree, and I will look into that. I don't like how there is just an undefined 8-byte header that comes from firmware, as that *should* be something in libstb-secvar for the exact reason of consistency between userspace and firmware.

It's up to you if you want to merge and take this now, or wait for the libstb-secvar change. We can always patch it out in a later release if need be.